spinach/brokers/redis_scripts/set_concurrency_keys.lua

```diff
@@ -0,0 +1,43 @@
+local max_concurrency_key = ARGV[1]
+local current_concurrency_key = ARGV[2]
+-- tasks to register starting at ARGV[3]
+
+
+local function contains(t, e)
+  return t[e]
+end
+
+local old_max_values = redis.call('hkeys', max_concurrency_key)
+local old_current_values = redis.call('hkeys', current_concurrency_key)
+
+local new_task_names = {}
+
+for i=3, #ARGV do
+    local task_json = ARGV[i]
+    local task = cjson.decode(task_json)
+    new_task_names[task["name"]] = true
+
+    -- Override max_concurrency whatever it is already set to, if
+    -- anything.
+    redis.call('hset', max_concurrency_key, task["name"], task["max_concurrency"])
```

Comment:
This fails for tasks with max_retry but no max_concurrency:

```
['{"max_concurrency": null, "max_retries": 10, "name": "compute", "periodicity": null, "queue": "spinach"}']
```

```
Traceback (most recent call last):
  File "/home/nicolas/dev/spinach/spinach/utils.py", line 29, in run_forever
    func(*args, **kwargs)
  File "/home/nicolas/dev/spinach/spinach/engine.py", line 144, in _arbiter_func
    [task for task in self._tasks.tasks.values()]
  File "/home/nicolas/dev/spinach/spinach/brokers/redis.py", line 319, in set_concurrency_keys
    *_tasks,
  File "/home/nicolas/dev/spinach/spinach/brokers/redis.py", line 95, in _run_script
    return script(args=args)
  File "/home/nicolas/dev/spinach/venv/lib/python3.7/site-packages/redis/client.py", line 4079, in __call__
    return client.evalsha(self.sha, len(keys), *args)
  File "/home/nicolas/dev/spinach/venv/lib/python3.7/site-packages/redis/client.py", line 3143, in evalsha
    return self.execute_command('EVALSHA', sha, numkeys, *keys_and_args)
  File "/home/nicolas/dev/spinach/venv/lib/python3.7/site-packages/redis/client.py", line 901, in execute_command
    return self.parse_response(conn, command_name, **options)
  File "/home/nicolas/dev/spinach/venv/lib/python3.7/site-packages/redis/client.py", line 915, in parse_response
    response = connection.read_response()
  File "/home/nicolas/dev/spinach/venv/lib/python3.7/site-packages/redis/connection.py", line 756, in read_response
    raise response
redis.exceptions.ResponseError: Error running script (call to f_2627b9f71ac57670aaa14074c3170b5901428991): @user_script:22: @user_script: 22: Lua redis() command arguments must be strings or integers
```

Reply:
This was caused by the thinko you pointed out in the next comment.